api/src/infrastructure/docker.rs

```diff
@@ -641,6 +645,39 @@ impl Infrastructure for DockerInfrastructure {
             }
         }
     }
+
+    fn change_status(
+        &self,
+        app_name: &String,
+        service_name: &String,
+        status: ServiceStatus,
+    ) -> Result<Option<Service>, failure::Error> {
+        match self.get_app_container(app_name, service_name)? {
+            Some(container) => {
+                let docker = Docker::new();
+                let containers = docker.containers();
+                let c = containers.get(&container.id);
+
+                let mut runtime = Runtime::new()?;
+                let details = runtime.block_on(c.inspect())?;
+                match status {
+                    ServiceStatus::Running => {
+                        if !details.state.running {
+                            runtime.block_on(c.start())?;
+                        }
+                    }
+                    ServiceStatus::Paused => {
+                        if details.state.running {
```

Comment:
Does this mean that updates to the container could be lost, if the Web UI is toggled twice in quick succession, before Docker has implemented the change?

Reply:
The code now checks if Docker's REST API return 304 and if so the frontend will received also 202. Other error will be propagated through an 500.